RubeesEat.UnitTests/Model/InMemoryBillRepositoryTests.cs

```diff
@@ -13,25 +13,206 @@ public void GetAll_EntriesInDictionary_ReturnsEntries()
+        [
+            new EntryLine(persons[0], 500),
+            new EntryLine(persons[0], -300),
+            new EntryLine(persons[1], -200),
+        ]);
+        billRepository.Add(firstBill);
+
+        var entryLine1 = new EntryLine(persons[0], 1_000_000);
+        var entryLine2 = new EntryLine(persons[2], -500_000);
+        var entryLine3 = new EntryLine(persons[1], -500_000);
+        billRepository.Add(new Bill(Guid.NewGuid(), dateTime, description, [entryLine1, entryLine2, entryLine3]));
+        billRepository.Add(new Bill(Guid.NewGuid(), dateTime, description, [entryLine1, entryLine2, entryLine3]));
+        billRepository.Add(new Bill(Guid.NewGuid(), dateTime, description, [entryLine1, entryLine2, entryLine3]));
+        billRepository.Add(new Bill(Guid.NewGuid(), dateTime, description, [entryLine1, entryLine2, entryLine3]));
+
+        var balanceChangesFromUser = billRepository.GetRecentBalanceChanges(defaultUserGuid, 3);
+
+        Assert.That(balanceChangesFromUser.Count, Is.EqualTo(3));
+    }
+
+    [Test]
+    public void GetRecentBalanceChanges_ZeroBalanceChangesFromUser_ReturnsEmpty()
+    {
+        var defaultUserGuid = Guid.NewGuid();
+        Person defaultUser = new Person(defaultUserGuid, "DefaultFirstName", "DefaultLastName");
+        var billRepository = new InMemoryBillRepository();
+
+        var personRepository = new InMemoryPersonRepository(defaultUser.Id);
+        personRepository.Add(defaultUser);
+        personRepository.Add(new Person(Guid.NewGuid(), "Item", "Arslan"));
+        personRepository.Add(new Person(Guid.NewGuid(), "Patrick", "Widener"));
+        var persons = personRepository.GetAll().ToImmutableList();
+        DateTime firstDate = new DateTime(2024, 9, 9);
+        string firstDescription = "Mittagessen auf Patrick sein Nacken";
+        var firstBill = new Bill(Guid.NewGuid(), firstDate, firstDescription,
+        [
+            new EntryLine(persons[2], 500),
+            new EntryLine(persons[2], -300),
+            new EntryLine(persons[1], -200),
+        ]);
+        billRepository.Add(firstBill);
+        billRepository.Add(new Bill(Guid.NewGuid(), new DateTime(2024, 9, 9), "Mittagessen auf Patrick sein Nacken",
+        [
+            new EntryLine(persons[1], 1_000_000),
+            new EntryLine(persons[2], -500_000),
+            new EntryLine(persons[1], -500_000)
+        ]));
+        billRepository.Add(new Bill(Guid.NewGuid(), new DateTime(2024, 9, 9), "Mittagessen auf Patrick sein Nacken",
+        [
+            new EntryLine(persons[1], 1_000_000),
+            new EntryLine(persons[2], -500_000),
+            new EntryLine(persons[1], -500_000)
+        ]));
+
+        var balanceChangesFromUser = billRepository.GetRecentBalanceChanges(defaultUserGuid, 1);
+
+        Assert.That(balanceChangesFromUser.Count, Is.EqualTo(0));
+    }
+
+    [Test]
+    public void GetAllFromUser_EntriesExist_OrderedByDate()
```

Comment:
I think this test method is not named correctly as it tests `GetRecentBalanceChanges` and not `GetAllFromUser`